__tests__/app.test.js

```diff
@@ -127,6 +127,27 @@ describe("GET /api/articles", () => {
         
         })
     });
+    test('200: responds with an array of articles filtered by topic query', () => {
+        return request(app)
+        .get("/api/articles?topic=mitch")
+        .expect(200)
+        .then(({ body }) => {
+            expect(Array.isArray(body.articles)).toBe(true)
+            expect(body.articles.length).toBe(12)
+            body.articles.forEach(article => {
+                expect(article.topic).toBe('mitch')
+            });
+        });
+    });
+    test('200: responds with an empty array if no articles match the topic', () => {
+        return request(app)
+            .get("/api/articles?topic=northcoders")
+            .expect(200)
+            .then(({ body }) => {
+                expect(Array.isArray(body.articles)).toBe(true);
+                expect(body.articles.length).toBe(0);
+            });
+    });
```

Comment:
Good testing so far, but we need to do a little bit more consideration here, think about the following...
What if a topic doesn't exist? We'd expect a 404 here
What if a topic does exist but doesn't have any articles? This should be a 200

We'd need to query the topics table to find out if what has been passed in is valid or notes then use that info to determine our response, give this a shot!